cdc/sinkv2/eventsink/event_appender.go

```diff
@@ -42,10 +46,51 @@ var _ Appender[*model.SingleTableTxn] = (*TxnEventAppender)(nil)
 type TxnEventAppender struct{}
 
 // Append appends the given rows to the given txn buffer.
+// The callers of this function should **make sure** that
+// the commitTs and startTs of rows is **strictly increasing**.
+// 1. Txns ordered by commitTs and startTs.
+// 2. Rows grouped by startTs and big txn batch,
+// cause the StartTs is the unique identifier of the transaction.
 func (t *TxnEventAppender) Append(
 	buffer []*model.SingleTableTxn,
 	rows ...*model.RowChangedEvent,
 ) []*model.SingleTableTxn {
-	// TODO implement me
-	panic("implement me")
+	for _, row := range rows {
+		noTxn := len(buffer) == 0
+
+		if noTxn ||
+			// Normally, this means the commitTs grows.
+			buffer[len(buffer)-1].GetCommitTs() != row.CommitTs ||
+			// Normally, this means we meet a new big txn batch.
+			row.SplitTxn ||
+			// Normally, this means we meet a new txn.
+			buffer[len(buffer)-1].StartTs < row.StartTs {
+			// fail-fast check
+			commitTsDecreased := !noTxn &&
+				buffer[len(buffer)-1].GetCommitTs() > row.CommitTs
+			if commitTsDecreased {
+				log.Panic("The commitTs of the emit row is less than the received row",
+					zap.Uint64("lastReceivedCommitTs", buffer[len(buffer)-1].GetCommitTs()),
+					zap.Any("row", row))
+			}
+			startTsDecreased := !noTxn &&
+				!row.SplitTxn && buffer[len(buffer)-1].StartTs > row.StartTs
+			if startTsDecreased {
+				log.Panic("The startTs of the emit row is less than the received row",
+					zap.Any("lastReceivedStartTs", buffer[len(buffer)-1].GetCommitTs()),
+					zap.Any("row", row))
+			}
+
+			buffer = append(buffer, &model.SingleTableTxn{
+				StartTs:   row.StartTs,
+				CommitTs:  row.CommitTs,
+				Table:     row.Table,
+				ReplicaID: row.ReplicaID,
```

Comment:
Where will this field be used? Whether it should be deleted?

Reply:
You mean `ReplicaID`?

Reply:
It looks like it's useless, but I don't know if there will be compatibility issues if I delete it directly, I will try to delete it in another pr.

Reply:
replicaid seems to used for the dual-way replication which has been deleted

Reply:
yes, I'll try to remove it in next PR. I'm not sure now if a direct deletion would introduce compatibility issues as it would be coded.